tests/testnoise.nim

```diff
@@ -0,0 +1,251 @@
+      asyncCheck await transport1.listen(server, connHandler)
+
+      let
+        transport2: TcpTransport = newTransport(TcpTransport)
+        clientInfo = PeerInfo.init(PrivateKey.random(RSA), [transport1.ma])
+        clientNoise = newNoise(clientInfo.privateKey, outgoing = true)
+        conn = await transport2.dial(transport1.ma)
+        sconn = await clientNoise.secure(conn)
+
+        msg = await sconn.read(6)
+
+      await sconn.close()
+      await transport1.close()
+
+      result = cast[string](msg) == "Hello!"
+
+    check:
+      waitFor(testListenerDialer()) == true
+
+  test "e2e: handle read + noise":
+    proc testListenerDialer(): Future[bool] {.async.} =
+      let
+        server: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+        serverInfo = PeerInfo.init(PrivateKey.random(RSA), [server])
+        serverNoise = newNoise(serverInfo.privateKey, outgoing = false)
+        readTask = newFuture[void]()
+
+      proc connHandler(conn: Connection) {.async, gcsafe.} =
+        let sconn = await serverNoise.secure(conn)
+        defer:
+          await sconn.close()
+        let msg = await sconn.read(6)
+        check cast[string](msg) == "Hello!"
+        readTask.complete()
+
+      let
+        transport1: TcpTransport = newTransport(TcpTransport)
+      asyncCheck await transport1.listen(server, connHandler)
+
+      let
+        transport2: TcpTransport = newTransport(TcpTransport)
+        clientInfo = PeerInfo.init(PrivateKey.random(RSA), [transport1.ma])
+        clientNoise = newNoise(clientInfo.privateKey, outgoing = true)
+        conn = await transport2.dial(transport1.ma)
+        sconn = await clientNoise.secure(conn)
+
+      await sconn.write("Hello!".cstring, 6)
+      await readTask
+      await sconn.close()
+      await transport1.close()
+
+      result = true
+
+    check:
+      waitFor(testListenerDialer()) == true
+
+  test "e2e use switch dial proto string":
+    proc testSwitch(): Future[bool] {.async, gcsafe.} =
+      let ma1: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+      let ma2: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+
+      var peerInfo1, peerInfo2: PeerInfo
+      var switch1, switch2: Switch
+      var awaiters: seq[Future[void]]
+
+      (switch1, peerInfo1) = createSwitch(ma1, false)
+
+      let testProto = new TestProto
+      testProto.init()
+      testProto.codec = TestCodec
+      switch1.mount(testProto)
+      (switch2, peerInfo2) = createSwitch(ma2, true)
+      awaiters.add(await switch1.start())
+      awaiters.add(await switch2.start())
+      let conn = await switch2.dial(switch1.peerInfo, TestCodec)
+      await conn.writeLp("Hello!")
+      let msg = cast[string](await conn.readLp())
+      check "Hello!" == msg
+
+      await allFutures(switch1.stop(), switch2.stop())
+      await allFutures(awaiters)
+      result = true
+
+    check:
+      waitFor(testSwitch()) == true
+
+  # test "interop with rust noise":
```

Comment:
are we  now able to interop with rust, are this tests supposed to be here?

Reply:
we can, it's disabled tho cos in CI there is no rust environment
Any opinion on this btw? should we actually load up rust env and test in CI?

Reply:
It would be definitely useful on many levels, but I'm not sure how hard would it be to setup a full rust pipeline. Another thing is that it would be nice to test all parts of the stack against rust to ensure full compatibility not just noise, but rust lacks (AFAIK) a daemon so we can't setup anything like we already have for the go interop. In any case, we can deal with that in a separate issue.